src/Blocks/Table.purs

```diff
@@ -0,0 +1,87 @@
+module Ocelot.Blocks.Table where
+
+import Prelude
+
+import DOM.HTML.Indexed (HTMLtable, HTMLtbody, HTMLtd, HTMLth, HTMLthead, HTMLtr)
+import Halogen.HTML (HTML(..), IProp)
+import Halogen.HTML as HH
+import Halogen.HTML.Properties as HP
+import Ocelot.Block.Builder (blockBuilder)
+import Ocelot.HTML.Properties ((<&>))
+
+tableClasses :: Array HH.ClassName
+tableClasses = HH.ClassName <$>
+  [ "w-full"
+  , "text-center"
+  , "border-collapse"
+  ]
+
+table
+  :: ∀ p i
+   . Array (IProp HTMLtable i)
+  -> Array (HTML p i)
+  -> HTML p i
+table = blockBuilder HH.table tableClasses
+
+table_
+  :: ∀ p i
+   . Array (HTML p i)
+  -> HTML p i
+table_ = table []
+
+row
+  :: ∀ p i
+   . Array (IProp HTMLtr i)
+  -> Array (HTML p i)
+  -> HTML p i
+row = HH.tr
+
+row_
+  :: ∀ p i
+   . Array (HTML p i)
+  -> HTML p i
+row_ = HH.tr_
+
+headerClasses :: Array HH.ClassName
+headerClasses = HH.ClassName <$>
+  [ "bg-grey-90"
+  , "py-4"
+  , "px-5"
+  , "font-medium"
+  , "text-black-20"
+  ]
+
+header
+  :: ∀ p i
+   . Array (IProp HTMLth i)
+  -> Array (HTML p i)
+  -> HTML p i
+header = blockBuilder HH.th headerClasses
```

Comment:
As you can see, I opted to go with more descriptive names o `header` and below, `cell`, instead of just `th` and `td` which I find less helpful. Open to counter-arguments on this. I also felt it was unnecessary to include `thead` and `tbody` equivalents, as they're not even needed when defining your html.